def oops():
    raise IndexError("This is an IndexError raised by oops() function.")

def catcher():
    try:
        oops()
    except IndexError as e:
        print("Caught an exception:", e)

if __name__ == "__main__":
    catcher()
